apps/src/code-studio/pd/components/schoolAutocompleteDropdownWithCustomFields.jsx

```diff
@@ -0,0 +1,134 @@
+/*
+ * Autocomplete school dropdown with
+ * custom fields if school not found
+ */
+import React, {PropTypes} from 'react';
+import {FormGroup, Row, Col, ControlLabel, HelpBlock} from 'react-bootstrap';
+import SchoolAutocompleteDropdown from '@cdo/apps/templates/SchoolAutocompleteDropdown';
+import CustomSchoolInfo from './customSchoolInfo';
+import {isZipCode} from '@cdo/apps/util/formatValidation';
+
+
+const VALIDATION_STATE_ERROR = "error";
+const OTHER_SCHOOL_VALUE = "-1";
+
+export default class SchoolAutocompleteDropdownWithCustomFields extends React.Component {
+  static propTypes = {
+    school_id: PropTypes.string,
+    school_info: PropTypes.object,
+    onSchoolInfoChange: PropTypes.func,
+    errors: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      school_id: null,
+      school_info: this.props.school_info,
+      errors: this.props.errors || {}
+    };
+  }
+
+  updateOnSchoolInfoChange = (school_info) => {
+    this.setState(school_info);
+    if (this.props.onSchoolInfoChange) {
+      this.props.onSchoolInfoChange(school_info);
+    }
+  };
+
+  handleSchoolDropdownChange = (selection) => {
+    this.setState({school_id: selection && selection.value});
+    if (selection && selection.value !== OTHER_SCHOOL_VALUE) {
+      this.updateOnSchoolInfoChange({
+        school_info: {
+          school_id: selection.school.nces_id,
+          school_name: selection.school.name,
+          school_state: selection.school.state,
+          school_zip: selection.school.zip,
+          school_type: selection.school.school_type
+        }
+      });
+    } else {
+      this.updateOnSchoolInfoChange({school_info: {}});
+    }
+  };
+
+  render() {
+    return (
+      <div>
+        {this.state.school_id !== OTHER_SCHOOL_VALUE &&
+          <FormGroup
+            id="school_id"
+            validationState={this.props.errors.hasOwnProperty("school_id") ? VALIDATION_STATE_ERROR : null}
+          >
+            <Row>
+              <Col md={6}>
+                <ControlLabel>
+                  School
+                  <span className="form-required-field"> *</span>
+                </ControlLabel>
+              </Col>
+            </Row>
+            <Row>
+              <Col md={12}>
+                <SchoolAutocompleteDropdown
+                  value={this.state.school_id}
+                  onChange={this.handleSchoolDropdownChange}
+                />
+              </Col>
+            </Row>
+            <HelpBlock>{this.props.errors.school_id}</HelpBlock>
+          </FormGroup>
+        }
+        {this.state.school_id && this.state.school_id === OTHER_SCHOOL_VALUE &&
+          <CustomSchoolInfo
+            school_info={this.props.school_info}
+            onSchoolInfoChange={this.props.onSchoolInfoChange}
+            errors={this.props.errors}
+          />
+        }
+      </div>
+    );
+  }
+
+  static getSchoolInfoErrors(school_info) {
```

Comment:
nit: I prefer `validate...` over `get...Errors`. It more clearly explains the primary purpose of the method. Add a header comment describing the return format.